testutil/keeper/unit_test_helpers.go

```diff
@@ -209,13 +209,18 @@ func GetNewVSCMaturedPacketData() types.VSCMaturedPacketData {
 // SetupForStoppingConsumerChain registers expected mock calls and corresponding state setup
 // which asserts that a consumer chain was properly stopped from StopConsumerChain().
 func SetupForStoppingConsumerChain(t *testing.T, ctx sdk.Context,
-	providerKeeper *providerkeeper.Keeper, mocks MockedKeepers,
+	providerKeeper *providerkeeper.Keeper, mocks MockedKeepers, closeChan bool,
 ) {
 	t.Helper()
 	expectations := GetMocksForCreateConsumerClient(ctx, &mocks,
 		"chainID", clienttypes.NewHeight(4, 5))
 	expectations = append(expectations, GetMocksForSetConsumerChain(ctx, &mocks, "chainID")...)
-	expectations = append(expectations, GetMocksForStopConsumerChain(ctx, &mocks)...)
+
+	// StopConsumerChain receives a `closeChan` parameter. If the `closeChan` argument is true, then
+	// we register additional expected mock calls that capture the fact that the underlying channel is closed.
+	if closeChan {
```

Comment:
Note that when `closeChan` is false, we're not mocking anything for  `StopConsumerChain`, as the name (SetupForStoppingConsumerChain) suggests. This method becomes a more generic setup method which mocks and calls `CreateConsumerClient` and `SetConsumerChain`.  

Maybe we refactor to make naming more clear about what's going on under the hood.

Reply:
Good catch! I rewrote the godoc of this function to clearly state this.

> Maybe we refactor to split things out and/or make naming more accurate

I agree. It would be nice to have 2 functions here: i) `SetupForStoppingConsumerChain` that just sets up the consumer chain for stopping, ii) `TestChannelIsClosed` that asserts that the underlying channel is closed when we call `StopConsumerChain` with `closeChan` being `true`. Then the idea would be that in a test we do something like the following:
```
SetupForStoppingConsumerChain(...)
StopConsumerChain(...) // e.g., we call to stop the consumer chain through `OnTimeoutPacket`
TestChannelIsClosed(...)
TestProviderStateIsCleanedAfterConsumerChainIsStopped(...) // tests that the provider's chain is closed and hence is used as a proxy to verify that the chain was closed 
```
where `TestChannelIsClosed` would need to set the expectations from [GetMocksForStopConsumerChain](https://github.com/cosmos/interchain-security/blob/cca008d856e3ffc60ec1a486871d0faa702abe26/testutil/keeper/expectations.go#L85), so something like:
```
func TestChannelIsClosed(...) {
    ....
    mocks.MockChannelKeeper.EXPECT().ChanCloseInit(...).Times(1)
}
```
The issue with the above approach is that expectations are set after the `StopConsumerChain` call and hence when `StopConsumerChain` is called we would fail with something akin to `Unexpected call to ChanCloseInit([1])`.
To solve the issue, this PR introduces the `SetupForStoppingConsumerChainChannel` function and does  
```
SetupForStoppingConsumerChain(...)
SetupForStoppingConsumerChainChannel(...)
StopConsumerChain(...)
TestProviderStateIsCleanedAfterConsumerChainIsStopped(...)
```
The above approach might not be that beautiful since we do testing before (i.e., test that the channel would be closed) and after (i.e., that the producer's chain is cleaned) the `StopConsumerChain`. Nevertheless, this approachs feels the nicest way to circumvent the mock expectations issue I mentioned above.